ts/features/bonus/cgn/screens/activation/CgnActivationLoadingScreen.tsx

```diff
@@ -1,29 +1,41 @@
 import * as React from "react";
 import { connect } from "react-redux";
-import { View } from "native-base";
 import { GlobalState } from "../../../../../store/reducers/types";
 import { Dispatch } from "../../../../../store/actions/types";
-import { H1 } from "../../../../../components/core/typography/H1";
-import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
+import { isCgnActivationLoading } from "../../store/reducers/activation";
+import {
+  cgnActivationCancel,
+  cgnRequestActivation
+} from "../../store/actions/activation";
+import { LoadingErrorComponent } from "../../../bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
+import I18n from "../../../../../i18n";
 
 type Props = ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps>;
 
 /**
  * Screen which is displayed when a user requested a CGN activation and we are waiting for the result from the backend
  */
-const CgnActivationLoadingScreen = (_: Props): React.ReactElement => (
-  // PLACEHOLDER for the loading error component screen
-  <BaseScreenComponent>
-    <View>
-      <H1>{"Activation is loading"}</H1>
-    </View>
-  </BaseScreenComponent>
+const CgnActivationLoadingScreen = (props: Props): React.ReactElement => (
+  <LoadingErrorComponent
+    isLoading={props.isLoading}
+    onRetry={props.onRetry}
+    loadingCaption={I18n.t("bonus.cgn.activation.loading.caption")}
+    loadingSubtitle={I18n.t("bonus.cgn.activation.loading.subCaption")}
+    errorText={I18n.t("bonus.cgn.activation.error.title")}
+    errorSubText={I18n.t("bonus.cgn.activation.error.body")}
+    onAbort={props.onCancel}
+  />
 );
 
-const mapStateToProps = (_: GlobalState) => ({});
+const mapStateToProps = (state: GlobalState) => ({
+  isLoading: isCgnActivationLoading(state)
```

Comment:
With this condition, after the first error, if the user choose to retry it will not see the loading screen because the value in the store always remains the same (in the store section `bonus.cgn.activation.status` value will be `ERROR`).

https://user-images.githubusercontent.com/26501317/108886246-08d6a500-7609-11eb-9888-4f6992b8148b.mov

in the attached video I added a consistent delay to the dev server and made the api return error.
How about making the loading condition based on a section of the store that takes into account the start of a new request and its outcome?

Reply:
Ok I got the point, this store section is yet used only for Activation progress state, after the implementation of #2812 we have another API for the CGN activation status polling


What about setting the activation state to
```typescript     
      {
        ...state,
        status: CgnActivationProgressEnum.PROGRESS
      };
```

when we dispatch the `cgnRequestActivation.request` action?

Reply:
Pushed this solution in ff0cff3

And now it shows back the loader when retrying the activation